src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -139,14 +168,21 @@ private static Long getTheme(String widget) {
     }
 
     private static native void paintBackground(int[] buffer, long theme,
-                                               int part, int state, int x,
-                                               int y, int w, int h, int stride);
+                                               int part, int state, int rectRight,
+                                               int rectBottom, int w, int h, int stride);
 
     public static void paintBackground(int[] buffer, String widget,
-           int part, int state, int x, int y, int w, int h, int stride) {
+           int part, int state, int x, int y, int w, int h, int stride, int dpi) {
         readLock.lock();
         try {
-            paintBackground(buffer, getTheme(widget), part, state, x, y, w, h, stride);
+
+            /* We get the part size vased on the theme for current screen DPI and pass it to paintBackground */
+            Dimension d = getPartSize(getTheme(widget, dpi), part, state);
+            paintBackground(buffer,
+                            getTheme(widget, dpi),
+                            part, state,
+                            (int)d.getWidth(),
+                            (int)d.getHeight() , w, h, stride);
```

Comment:
```suggestion
                            d.width,
                            d.height, w, h, stride);
```
You can access the fields directly and avoid casting.